engine/src/main/scala/cromwell/engine/backend/sge/SgeBackend.scala

```diff
@@ -219,6 +223,16 @@ case class SgeBackend(actorSystem: ActorSystem) extends Backend with SharedFileS
   override def executionInfoKeys: List[String] = List(InfoKeys.JobNumber)
 
   override def callEngineFunctions(descriptor: BackendCallJobDescriptor): CallEngineFunctions = {
-    new SgeCallEngineFunctions(descriptor.workflowDescriptor.ioManager, buildCallContext(descriptor))
+    new SgeCallEngineFunctions(descriptor.workflowDescriptor.fileSystems, buildCallContext(descriptor))
+  }
+
+  override def fileSystems(options: WorkflowOptions, workflowRootPath: String): List[FileSystem] = {
+    // TODO Should GCS operations have their own execution context ?
+    implicit val executionContext = scala.concurrent.ExecutionContext.global
+
+    val gcsStorage = StorageFactory.userAuthenticated(options) orElse StorageFactory.cromwellAuthenticated
+    val gcs = gcsStorage map GcsFileSystemProvider.apply map { _.getDefaultFileSystem } toOption
+
+    List(gcs, Option(defaultFileSystem)).flatten
```

Comment:
> FilesSystems feel separate than Backends

Yes, but this is a bigger change than the primary purpose of this ticket (which this PR is already over reaching anyway..)